modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -0,0 +1,506 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TriPinHexAssemblyGenerator.h"
+#include "libmesh/mesh_smoother_laplace.h"
+
+#include <cmath>
+
+registerMooseObject("ReactorApp", TriPinHexAssemblyGenerator);
+
+InputParameters
+TriPinHexAssemblyGenerator::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "num_sectors_per_side", "num_sectors_per_side>0", "Number of azimuthal sectors per side.");
+  params.addRangeCheckedParam<unsigned int>(
+      "background_intervals",
+      1,
+      "background_intervals>0",
+      "Number of radial meshing intervals in background region (area outside the rings).");
+  params.addRangeCheckedParam<std::vector<subdomain_id_type>>(
+      "background_block_ids",
+      "background_block_ids>0",
+      "Optional block ids for the background region.");
+  params.addParam<std::vector<SubdomainName>>("background_block_names",
+                                              "Optional block names for the background region.");
+  params.addRangeCheckedParam<std::vector<std::vector<Real>>>(
+      "ring_radii", "ring_radii>0", "Radii of the three sets of major concentric circles (rings).");
+  params.addRangeCheckedParam<std::vector<std::vector<unsigned int>>>(
+      "ring_intervals",
+      "ring_intervals>0",
+      "Number of radial mesh intervals within each set of major concentric circles.");
+  params.addRangeCheckedParam<std::vector<std::vector<subdomain_id_type>>>(
+      "ring_block_ids", "ring_block_ids>0", "Optional block ids for the ring regions.");
+  params.addParam<std::vector<std::vector<SubdomainName>>>(
+      "ring_block_names", "Optional block names for the ring regions.");
+  MooseEnum hexagon_size_style("apothem radius", "radius");
+  params.addParam<MooseEnum>("hexagon_size_style",
+                             hexagon_size_style,
+                             "Style in which hexagon size is given (apothem = center to face, "
+                             "radius = center to vertex). Options: " +
+                                 hexagon_size_style.getRawNames());
```

Comment:
```suggestion
                             "radius = center to vertex).");
```
no need to list them, docs already do list options for an enum